syntax = "proto3";

message Job{
  message Start{
    uint64 timestamp = 1;
    string cipher = 2;
    string hash = 3;
    string transaction_id = 4;
  }

  message End{
    uint64 timestamp = 1;
    string transaction_id = 2;
  }
  string company_public_key = 1;
  string candidate_public_key = 2;
  string job_id = 3;
  Start start = 4;
  End end = 5;

}

message JobContainer{
  repeated Job entries = 1;
}